frontend/app/routes/protected/person-case/parent-details.tsx

```diff
@@ -0,0 +1,424 @@
+      }
+
+      (context.session.inPersonSINCase ??= {}).parentDetails = parseResult.output;
+      throw i18nRedirect('routes/protected/person-case/previous-sin.tsx', request);
+    }
+
+    default: {
+      throw new AppError(`Unrecognized action: ${action}`, ErrorCodes.UNRECOGNIZED_ACTION);
+    }
+  }
+}
+
+export default function CreateRequest({ loaderData, actionData, params }: Route.ComponentProps) {
+  const { t } = useTranslation(handle.i18nNamespace);
+  const fetcherKey = useId();
+  const fetcher = useFetcher<Info['actionData']>({ key: fetcherKey });
+  const isSubmitting = fetcher.state !== 'idle';
+  const errors = fetcher.data?.errors;
+  const defaultFormValues = loaderData.defaultFormValues;
+
+  return (
+    <>
+      <PageTitle subTitle={t('protected:in-person.title')}>{t('protected:parent-details.page-title')}</PageTitle>
+      <FetcherErrorSummary fetcherKey={fetcherKey}>
+        <fetcher.Form method="post" noValidate>
+          <ParentInformation defaultFormValues={defaultFormValues} errors={errors} />
+          <div className="mt-8 flex flex-row-reverse flex-wrap items-center justify-end gap-3">
+            <Button name="action" value="next" variant="primary" id="continue-button" disabled={isSubmitting}>
+              {t('protected:person-case.next')}
+            </Button>
+            <Button name="action" value="back" id="back-button" disabled={isSubmitting}>
+              {t('protected:person-case.previous')}
+            </Button>
+          </div>
+        </fetcher.Form>
+      </FetcherErrorSummary>
+    </>
+  );
+}
+
+type FormData = {
+  unavailable?: boolean;
+  givenName?: string;
+  lastName?: string;
+  country?: string;
+  province?: string;
+  city?: string;
+};
+
+interface ParentInformationProps {
+  defaultFormValues: FormData[];
+  errors?: Record<string, [string, ...string[]] | undefined>;
+}
+
+function ParentInformation({ defaultFormValues, errors }: ParentInformationProps) {
+  const { t } = useTranslation(handle.i18nNamespace);
+  const [parentAmount, setParentAmount] = useState(Math.max(defaultFormValues.length, 1));
+  const { getId, removeId } = useIdList();
+
+  const maxParents = 4;
+  const canAddParent = parentAmount < maxParents;
+
+  const onAddParent = () => {
+    if (canAddParent) setParentAmount(parentAmount + 1);
+  };
+  const onCloseParent = (index: number) => {
+    defaultFormValues.splice(index, 1);
+    removeId(index);
+    setParentAmount(parentAmount - 1);
+  };
+
+  return (
+    <>
+      <input type="hidden" name="parent-amount" value={parentAmount} />
+      <div className="space-y-10">
+        {Array.from({ length: parentAmount }).map((_, index) => (
+          <ParentForm
+            key={getId(index)}
+            index={index}
+            defaultValues={defaultFormValues[index]}
+            errors={errors}
+            onClose={parentAmount > 1 ? onCloseParent : undefined}
```

Comment:
I don't like the name `onCloseParent` here. It really doesn't describe what is happening (in my opinion). You're not closing a parent, you're removing it, so it should be `removeParent`. Also, the index should be passed in here so it is clear what is happening.

I think it would be better to use something like:

``` tsx
onClose={parentAmount > 1 ? () => removeParent(index) : undefined}
```